extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/FluoQueryMetadataCache.java

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    @Override
+    public ProjectionMetadata readProjectionMetadata(SnapshotBase tx, String nodeId) {
+        Optional<NodeType> type = NodeType.fromNodeId(nodeId);
+        checkArgument(type.isPresent() && type.get() == NodeType.PROJECTION);
+        LOG.debug("Retrieving Metadata from Cache: {}", nodeId);
+        try {
+            return (ProjectionMetadata) commonNodeMetadataCache.get(nodeId, new Callable<CommonNodeMetadata>() {
+                @Override
+                public CommonNodeMetadata call() throws Exception {
+                    LOG.debug("Seeking Metadata from Fluo Table: {}.", nodeId);
+                    return dao.readProjectionMetadata(tx, nodeId);
+                }
+            });
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to access ProjectionMetadata for nodeId: " + nodeId, e);
+        }
+    }
+
+    @Override
+    public AggregationMetadata readAggregationMetadata(SnapshotBase tx, String nodeId) {
+        Optional<NodeType> type = NodeType.fromNodeId(nodeId);
+        try {
+            checkArgument(type.isPresent() && type.get() == NodeType.AGGREGATION);
+            LOG.debug("Retrieving Metadata from Cache: {}", nodeId);
+            return (AggregationMetadata) commonNodeMetadataCache.get(nodeId, new Callable<CommonNodeMetadata>() {
+                @Override
+                public CommonNodeMetadata call() throws Exception {
+                    LOG.debug("Seeking Metadata from Fluo Table: {}.", nodeId);
+                    return dao.readAggregationMetadata(tx, nodeId);
+                }
+            });
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to access AggregationMetadata for nodeId: " + nodeId, e);
+        }
+    }
+
+    @Override
+    public ConstructQueryMetadata readConstructQueryMetadata(SnapshotBase tx, String nodeId) {
+        Optional<NodeType> type = NodeType.fromNodeId(nodeId);
+        try {
+            checkArgument(type.isPresent() && type.get() == NodeType.CONSTRUCT);
+            LOG.debug("Retrieving Metadata from Cache: {}", nodeId);
+            return (ConstructQueryMetadata) commonNodeMetadataCache.get(nodeId, new Callable<CommonNodeMetadata>() {
+                @Override
+                public CommonNodeMetadata call() throws Exception {
+                    LOG.debug("Seeking Metadata from Fluo Table: {}.", nodeId);
+                    return dao.readConstructQueryMetadata(tx, nodeId);
+                }
+            });
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to access ConstructQueryMetadata for nodeId: " + nodeId, e);
+        }
+    }
+
+    @Override
+    public PeriodicQueryMetadata readPeriodicQueryMetadata(SnapshotBase tx, String nodeId) {
+        Optional<NodeType> type = NodeType.fromNodeId(nodeId);
+        try {
+            checkArgument(type.isPresent() && type.get() == NodeType.PERIODIC_QUERY);
```

Comment:
same....I'm just going to stop and wait to see if this actually gets caught

Reply:
Same as above.